src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -113,18 +115,28 @@ public static void ReadAndHashUntilPosition(BinaryReader reader, HashAlgorithm h
         /// <param name="hashAlgorithm">HashAlgorithm used to hash contents</param>
         /// <param name="bytes">Content to hash</param>
         public static void HashBytes(HashAlgorithm hashAlgorithm, byte[] bytes)
+        {
+            HashBytes(hashAlgorithm, bytes, bytes?.Length ?? 0);
+        }
+
+        internal static void HashBytes(HashAlgorithm hashAlgorithm, byte[] bytes, int count)
         {
             if (hashAlgorithm == null)
             {
                 throw new ArgumentNullException(nameof(hashAlgorithm));
             }
 
+            if (count <= 0 || bytes.Length < count)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+
```

Comment:
This needs to go after the `bytes` argument check below.  Why?  `bytes` could be `null` here.